src/components/Input/index.tsx

```diff
@@ -2,49 +2,60 @@
 import {
   DetailedHTMLProps,
   InputHTMLAttributes,
-  useRef,
-  useState
+  useCallback,
+  useState, SyntheticEvent
 } from "react";
 
 import {
   FontAwesomeIcon,
   FontAwesomeIconProps
 } from "@fortawesome/react-fontawesome";
 import { IconProp } from "@fortawesome/fontawesome-svg-core";
+import { IconDefinition} from "@fortawesome/fontawesome-svg-core";
 import styles from "./input.module.css";
 
 type UserInputProps = {
   icon?: Omit<FontAwesomeIconProps, "icon"> & { icon: IconProp };
+  label?:string; 
 } & DetailedHTMLProps<
   InputHTMLAttributes<HTMLInputElement>,
   HTMLInputElement
 >;
 
-const UserInput = ({ ...props }: UserInputProps) => {
-  const { icon } = props;
+function useInputValidation(initialValue: string = "") {
+	const [input, setInput] = useState(initialValue);
+
+	const validateInput = (value: string): void => {
+		// Validation logic here
+		setInput(value);
+	};
 
-  const [input, setInput] = useState("");
+	return [input, validateInput] as const;
+}
 
-  const validateInput = (value: string) => {
-    setInput(value);
-  };
+const UserInput = ({ ...props }: UserInputProps) => {
+	
+  const {icon, label, id, children, className} = props; 
+	
+  const [input, validateInput] = useInputValidation();
+  const onKeyUp = useCallback((event: SyntheticEvent) => validateInput((event.currentTarget as HTMLInputElement).value), [validateInput]);
 
   const [type, setHidden] = useState(props.type);
```

Comment:
This shouldn't be a state if the parent is controlling it => The parent would have the ùseState` and updates in the parent would modify that state value and feed it down to the type, This would actually be a lock if there's nothing in this function that modifies the value

ex:

```ts
 const PasswordInput() => { 
     [type, setType] = useState(false); 
     
     return <UserInput type={type? "text" : "password"} onClick={() => setType(!type)}/>
}
```